### PR TITLE
Add organizationId

### LINKED ISSUE #6: [FEATURE] - Allow for specifying the organization id
Allow the `organizationId` to be specified in the configuration. See https://platform.openai.com/docs/api-reference/organization-optional

Right now there is no way to specify this.

### PR DESCRIPTION
**Pull request by edeandrea**
Allow the organizationId to be specified in the configuration. See https://platform.openai.com/docs/api-reference/organization-optional

Right now there is no way to specify this.

Since this is also a header, I introduced a new `GenericHeaderInjector` and then also refactored `ApiKeyHeaderInjector` and `AuthorizationHeaderInjector` to use it. I didn't eliminate `ApiKeyHeaderInjector` and `AuthorizationHeaderInjector` because that would have been a breaking change.

I also added a few things to `.gitignore` that I noticed were added when I loaded the project into IntelliJ and ran a build locally.

Fixes #6

### CODE REVIEW

File: src/main/java/dev/ai4j/openai4j/GenericHeaderInjector.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import okhttp3.Interceptor;
+import okhttp3.Request.Builder;
+import okhttp3.Response;
+
+class GenericHeaderInjector implements Interceptor {
+	private final Map<String, String> headers = new HashMap<>();
```
Comment (Heezer) [inline src/main/java/dev/ai4j/openai4j/GenericHeaderInjector.java:13]: I'm curious about why indentation is 8?? 😄  Default in project here is 4.
Find the fresh `.prettierrc` file in the project root - the config for the formatting tool `Prettier`.

**Review by Heezer — COMMENTED** [review]
Thanks for your changes! See my comment about indentation and resolve the merge conflicts, most probably they are because my merge to master from me 😄

File: src/main/java/dev/ai4j/openai4j/GenericHeaderInjector.java (hunk 1 above)
Comment (edeandrea) [reply src/main/java/dev/ai4j/openai4j/GenericHeaderInjector.java:13]: No idea? My IDE is set up as indent by 2, so no idea where 8 came from....

**Comment by edeandrea** [comment]
Thanks for the comment @Heezer ! I've resolved the conflicts.

File: src/main/java/dev/ai4j/openai4j/GenericHeaderInjector.java (hunk 1 above)
Comment (edeandrea) [reply src/main/java/dev/ai4j/openai4j/GenericHeaderInjector.java:13]: It looks like it is indented by 4 to me, regardless of what editor I open it in...

image

Comment (Heezer) [reply src/main/java/dev/ai4j/openai4j/GenericHeaderInjector.java:13]: looks like there are tabs, just format with 4 spaces, that's it 😄

Comment (edeandrea) [reply src/main/java/dev/ai4j/openai4j/GenericHeaderInjector.java]: How's that?

Comment (Heezer) [reply src/main/java/dev/ai4j/openai4j/GenericHeaderInjector.java]: There must be related setting in every IDE for that. I'm already approving then.

**Review by Heezer — APPROVED** [review]

File: src/main/java/dev/ai4j/openai4j/GenericHeaderInjector.java (hunk 1 above)
Comment (Heezer) [reply src/main/java/dev/ai4j/openai4j/GenericHeaderInjector.java]: now it's fine!! 👍

**Comment by edeandrea** [comment]
Once this is released I'll incorporate it into https://github.com/langchain4j/langchain4j/issues/344

**Comment by ai-for-java** [comment]
HI @edeandrea thanks for this! BTW why removing `ApiKeyHeaderInjector` would be a breaking change?

**Comment by edeandrea** [comment]
Oh I guess it wouldn't be because it's not `public`. Just realized that now.

**Pull request merged**